TypeTreeGeneratorAPI/NativeAPI.cs

```diff
@@ -177,6 +173,39 @@ unsafe public static int TypeTreeGenerator_generateTypeTreeNodesRaw(IntPtr typeT
             }
         }
 
+        [UnmanagedCallersOnly(EntryPoint = "TypeTreeGenerator_getMonoBehaviorDefinitions")]
+        unsafe public static int TypeTreeGenerator_getMonoBehaviorDefinitions(IntPtr typeTreeGeneratorPtr, IntPtr arrAddrPtr, IntPtr arrLengthPtr)
+        {            
+            if (typeTreeGeneratorPtr == IntPtr.Zero )
+            {
+                return -1;
+            }
+            try
+            {
+                var typeTreeGenerator = (TypeTreeGenerator)GCHandle.FromIntPtr(typeTreeGeneratorPtr).Target!;
+
+                var typeNames = typeTreeGenerator.GetMonoBehaviourDefinitions();
+
+                var arrayLength = typeNames.Count;
+                var arrayPtr = Marshal.AllocCoTaskMem(Marshal.SizeOf<IntPtr>() * arrayLength * 2);                
+                for (int i = 0; i < arrayLength; i++)
+                {
+                    string module = typeNames[i].Module.Name;
+                    string fullName = typeNames[i].FullName;
+                    Marshal.WriteIntPtr(arrayPtr, (i * 2) * Marshal.SizeOf<IntPtr>(), Marshal.StringToCoTaskMemAnsi(module));
```

Comment:
This might result in errors if the names include non ansi ascii compatible characters.
StringToCoTaskMemUTF8( would be better.

I should use this for the json code.

Reply:
I changed the encoding system locally just now, so that the system's default encoding gets used to prevent possible issues.